### ISSUE TITLE
C++ Adam optimizer

### ISSUE THREAD

**Issue by meganset** [pytorch/pytorch#36060]
## 🐛 Bug

There are some minor labelling errors in the default options check in  _optim/adah.h_: 
```
     auto betas = defaults.betas();
     TORCH_CHECK(std::get<0>(betas) >= 0, "Invalid learning rate: ", std::get<0>(betas));
     TORCH_CHECK(std::get<1>(betas) >= 0, "Invalid learning rate: ", std::get<1>(betas));
     TORCH_CHECK(defaults.weight_decay() >= 0, "Invalid learning rate: ", defaults.weight_decay());
```
I am also seeing worse convergance on some simple GAN's with MNIST,
e.g. the tutorial example, using the newer adam vs the 1.4 version,
but I don't have a good way to reproduce this yet.
the GAN's produce noticeably less convincing digits than using the 1.4 version for the same settings and number of iterations.

cc @yf225 @vincentqb

**Comment by yf225** [comment]
C++ Adam optimizer's implementation was changed to be exactly the same as the Python version. I am curious does the Python version have similar convergence behavior, or C++ version is also worse than Python version?

On the labelling error, thanks a lot for the catch and yes we should fix it. Curious would you be interested in sending a PR? Also cc. @anjali411

**Comment by meganset** [comment]
I don't think the changed _adam_ implementation is to blame here.
I was tripping over the new betas option specification (from two separate scalars to a 2-element tuple), so ended up with a misspecified set of options.
I'm seeing similar convergence now on some simple GANs & conditional GANs.

I'll try to submit a PR to fix the check messages later this evening.
Thanks

**Comment by meganset** [comment]
Adam check messages fixed vi PR #36161
